### PR TITLE
111 python package

### PR DESCRIPTION
**Pull request by wenwenchenbosch**
solve #111  

You can find the test publisher in https://test.pypi.org/project/kuksa-viss-client/

### CODE REVIEW

**Comment by wenwenchenbosch** [comment]
@SebastianSchildt  bug, I mentioned today fixed => ready to review now.

**Comment by callalilychen** [comment]
jenkins of pr-merge failed, because of modification of `jenkinsfile`. real result see https://kuksaval.northeurope.cloudapp.azure.com/job/kuksaval-boschresearch/job/111_python_package/

**Comment by wenwenchenbosch** [comment]
jenkins of pr-merge failed, because of modification of jenkinsfile. real result see https://kuksaval.northeurope.cloudapp.azure.com/job/kuksaval-boschresearch/job/111_python_package/

**Comment by SebastianSchildt** [comment]
Checking all Python related stuff. Puzzled about gpsd_feeder, is it non-functional now?

File: kuksa_client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+RUN apk update && apk add git bash
+ADD . /kuksa.val
+WORKDIR /kuksa.val
+RUN pip install --upgrade pip build
+RUN python -m build
+
+FROM python:3.8-alpine
+
+COPY --from=build /kuksa.val/dist /kuksa_dist
+RUN pip install kuksa_dist/*.whl
```
Comment (SebastianSchildt) [inline kuksa_client/Dockerfile:22]: Optimization: Would be more effcient to install in "build" dev and only copy files. Here the wheel always is part of the docker...

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,36 @@
+[metadata]
+name = kuksa_client
+author = Sebastian Schildt, Naresh Nayak, Wenwen Chen
+author_email = sebastian.schildt@de.bosch.com, naresh.nayak@de.bosch.com, wenwen.chen@de.bosch.com
+description = kuksa.val python clients and SDK
```
Comment (SebastianSchildt) [inline setup.cfg:5]: I think the name __has to__  to include VISS, becasue that is what is implementing

kuksa_viss

or

kuksa_viss_client

also it

File: kuksa_client/Dockerfile (hunk 1 above)
Comment (wenwenchenbosch) [reply kuksa_client/Dockerfile:22]: I do not know to install in "build" dev. However, the kuksa_dist will be removed in the next line

Comment (SebastianSchildt) [reply kuksa_client/Dockerfile]: Docker is "like git", due to the layers. Not matter you delete it or not, it always takes up space in the final image.
See testclient docker for an example https://github.com/eclipse/kuksa.val/blob/master/clients/vss-testclient/Dockerfile all python packages are installed locally, and then just the folder is copied.

Comment (wenwenchenbosch) [reply kuksa_client/Dockerfile]: ok, I got it running now

**Review by SebastianSchildt — CHANGES_REQUESTED** [review]
Daaemon dockerfile crashes

```
$ docker run -it --rm -v /Users/sebastian/Downloads/newconfig:/config  -p 127.0.0.1:8090:8090 -e LOG_LEVEL=ALL amd64/kuksa-val:dc3704a
Starting kuksa.val
No VSS tree, initilaize with example
No server keys configured, initialize with example
No jwt key configured, initialize with example
kuksa.val server
Commit dc3704a from 2021-03-17T14:23:42+01:00
Read configs from config.ini
Log START
VERBOSE: Try reading JWT pub key from /config/certs/jwt.key.pub
terminate called after throwing an instance of 'boost::wrapexcept<boost::bad_any_cast>'
  what():  boost::bad_any_cast: failed conversion using boost::any_cast
Aborted
```

**Comment by SebastianSchildt** [comment]
may already be in master. Could be alpine problem... Will investiagte first

File: README.md [hunk 3]
```diff
@@ -99,51 +103,6 @@ if no other config file is specified using the command line option `-c/--config-
...
-
-![Alt text](./doc/pictures/testclient_basic.gif "test client usage")
-
-Using the testclient, it is also possible to update and extend the VSS data structure. More details can be found [here](./doc/liveUpdateVSSTree.md).
-
-
-### Feeder
-
-We provider currently the following feeders:
-- [DBC feeder](./clients/feeder/dbc2val)
```
Comment (SebastianSchildt) [inline README.md:137]: Links broken

Comment (wenwenchenbosch) [reply README.md:137]: this part is removed

Comment (SebastianSchildt) [reply README.md:137]: Sorry wrong location. Here: https://github.com/boschresearch/kuksa.val/tree/111_python_package/kuksa_feeders

File: kuksa_feeders/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Kuksa feeders
+![kuksa.val Logo](../doc/pictures/logo.png)
+
+We provider currently the following feeders:
+
+Name | Description
+---- | -----------
+[GPS feeder](./kuksa_feeders/gps2val) | GPS data source for `kuksa.val` server
```
Comment (wenwenchenbosch) [inline kuksa_feeders/README.md:8]:
```suggestion
[GPS feeder](./gps2val) | GPS data source for `kuksa.val` server
```

File: kuksa_feeders/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+[DBC feeder](./kuksa_feeders/dbc2val) | DBC feeder for `kuksa.val` server
```
Comment (wenwenchenbosch) [inline kuksa_feeders/README.md:9]:
```suggestion
[DBC feeder](./dbc2val) | DBC feeder for `kuksa.val` server
```

**Comment by SebastianSchildt** [comment]
I seem to habe some dependency problem with the python apckege

```
l$ pip3 install -i https://test.pypi.org/simple/ kuksa-viss-client
Looking in indexes: https://test.pypi.org/simple/
Collecting kuksa-viss-client
  Downloading https://test-files.pythonhosted.org/packages/85/5a/8b4f4b7f2f18518133e1d7c713e41492f56657c3239629d98056e02b49a1/kuksa_viss_client-0.1.7.post114-py3-none-any.whl (33 kB)
Collecting pygments~=2.5
  Downloading https://test-files.pythonhosted.org/packages/3a/80/a52c0a7c5939737c6dca75a831e89658ecb6f590fb7752ac777d221937b9/Pygments-2.8.1-py3-none-any.whl (983 kB)
     |████████████████████████████████| 983 kB 4.7 MB/s 
Requirement already satisfied: websockets~=8.1 in /usr/local/lib/python3.8/dist-packages (from kuksa-viss-client) (8.1)
ERROR: Could not find a version that satisfies the requirement importlib-metadata>=3.6 (from kuksa-viss-client) (from versions: none)
ERROR: No matching distribution found for importlib-metadata>=3.6 (from kuksa-viss-client)
```

any idea?

**Comment by SebastianSchildt** [comment]
Solved. It is because it only checks the test pypi, can be fixed by install like this

```
 pip3 install --extra-index-url  https://test.pypi.org/simple/ kuksa-viss-client
```

**Review by SebastianSchildt — APPROVED** [review]
Works now

**Pull request merged**